### ISSUE TITLE
train_model.py

### ISSUE THREAD

**Issue by temur3559** [temur3559/UltimateCryptoBotV2#28]
import numpy as np
import pandas as pd
import tensorflow as tf
from ta.momentum import RSIIndicator
from ta.trend import EMAIndicator, MACD
from sklearn.preprocessing import MinMaxScaler
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import LSTM, Dense
import os

def load_data(file_path, sequence_length=50):
    df = pd.read_csv(file_path)

    # Indikatorlar
    df['rsi'] = RSIIndicator(df['close']).rsi()
    df['ema'] = EMAIndicator(df['close']).ema_indicator()
    macd = MACD(df['close'])
    df['macd'] = macd.macd_diff()
    df.dropna(inplace=True)

    features = df[['rsi', 'ema', 'macd']].values
    target = df['close'].values

    scaler = MinMaxScaler()
    features_scaled = scaler.fit_transform(features)

    X, y = [], []
    for i in range(sequence_length, len(features_scaled)):
        X.append(features_scaled[i-sequence_length:i])
        y.append(target[i])

    return np.array(X), np.array(y), scaler

def build_model(input_shape):
    model = Sequential()
    model.add(LSTM(64, return_sequences=False, input_shape=input_shape))
    model.add(Dense(1))
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

def train_model():
    file_path = "data/historical_data.csv"  # ma'lumot joylashuvi
    X, y, scaler = load_data(file_path)
    model = build_model((X.shape[1], X.shape[2]))
    model.fit(X, y, epochs=10, batch_size=32)
    model.save("model/lstm_model.h5")
    print("[TRAIN] Model saqlandi: model/lstm_model.h5")

if __name__ == "__main__":
    train_model()
